mlir/lib/Conversion/AMDGPUToROCDL/AMDGPUToROCDL.cpp

```diff
@@ -454,6 +454,20 @@ static void wmmaPushOutputOperand(ConversionPatternRewriter &rewriter,
   }
 }
 
+/// Return true if `type` is the E5M2 variant of an 8-bit float that is
+/// supported by the `_bf8` instructions on the given `chipset`.
+static bool isNativeBf8(Chipset chipset, Type type) {
```

Comment:
```suggestion
static bool hasNativeBF8(Chipset chipset, Type type) {
```

Reply:
Since it's a query about the type, not about the chipset, I'm happier with an "is" form.  Since I'm wordy, how about isNativeBf8ForChipset(Type,Chipset)?

Reply:
I don't understand the distinction. The ISA and the supported types are the property of the chipset, no?

Reply:
Well, yes, but all chipsets have a native BF8 type -- FNUZ for the older ones, OCP for the newer -- so the answer to "has native BF8" is "yes".  Rather, we want to know "is this type the same as the chipset's native BF8 type?"  Also, in a context in which we're doing isa<T1,T2>() and isFloat(), "is" flows better.

Reply:
>  Also, in a context in which we're doing isa<T1,T2>() and isFloat(), "is" flows better.

The issue with chains of `.isType` is that it won't be optimized across multiple types and always performed one by one (with short circuiting). Our casting infra might not do it today, but in order to utilize faster 'group' isa, you have to use it across the codebase in the first place.

Reply:
>  "is this type the same as the chipset's native BF8 type?"
Thanks for the explanation, I see the difference now. Maybe something like this then `isSupportedByNativeBf8(Chipset, Type)`?

If you prefer the current name, could you add a comment that explains what the intention is (similar to how you explained it above)?

Reply:
An equivalent way of phrasing it, from where I'm standing, is "is this one of the types that the `_fp8` or `_bf8` instructions on the given `chipset` will be expecting"?

Reply:
Previous job would have called it "typeIsNativelySupportedBf8ForChipset(Type,Chipset)" -- we liked naming the parameters in the function name -- or "typeIsExpectedBf8ForChipset(Type,Chipset)".  As I wrote those out, I think I like "expected" better than "native" or "supported," but I could be persuaded.

Reply:
Hm ... `isImplementedBf8OnChipset()`?

But `Expected` works too